### PR TITLE
nydus-test: Replace storage backend with BACKEND_PROXY and adapts github action

### PR DESCRIPTION
**Pull request by changweige**
test_api is the only case suite being tested. Let's also replace its storage backend type.
In addition, fill backend proxy configurations into github action.

### CODE REVIEW

**Comment by yqleng1987** [comment]
@changweige , a new test job has been submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@changweige , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6697>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6697>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6697>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6697>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6697>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6697>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6697>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

File: .github/workflows/it.yml [hunk 1]
```diff
@@ -79,7 +81,9 @@ jobs:
             "registry": {
                 "registry_url": "localhost:5000",
                 "registry_namespace": "",
-                "registry_auth": "Y2h1YW5sYW5nLmdjdzoxcWF6MndzeA=="
+                "registry_auth": "Y2h1YW5sYW5nLmdjdzoxcWF6MndzeA==",
```
Comment (imeoer) [inline .github/workflows/it.yml:84]: Is this sensitive?

Comment (changweige) [reply .github/workflows/it.yml:84]: We can trim it though it should point to any public registry.

**Comment by yqleng1987** [comment]
@changweige , your pull request has been updated. A new test job will be submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@changweige , your test job has passed, and no need to test again.

**Review by imeoer — APPROVED** [review]

**Pull request merged**